### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by JimmyCYJ**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

https://github.com/istio/istio/issues/24473

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -161,10 +161,10 @@ require (
 	gopkg.in/square/go-jose.v2 v2.3.1
 	gopkg.in/yaml.v2 v2.3.0
 	helm.sh/helm/v3 v3.2.4
-	istio.io/api v0.0.0-20200714164416-42acf9095d43
+	istio.io/api v0.0.0-20200721175012-ae75c7e9ae26
```
Comment (howardjohn) [inline go.mod:164]: I am pretty sure this will fail without olivers PR https://github.com/istio/istio/pull/25677

File: istio.deps [hunk 2]
```diff
@@ -4,6 +4,6 @@
     "name": "PROXY_REPO_SHA",
     "repoName": "proxy",
     "file": "",
-    "lastStableSHA": "47d38c13e18d78f8079d602feba218138ce0fdab"
+    "lastStableSHA": "42c7f6d3c4328b5ae318421c7b332f48ccfe616b"
```
Comment (howardjohn) [inline istio.deps:7]: this will fail without https://github.com/istio/istio/pull/25705

File: go.mod (hunk 1 above)
Comment (jwendell) [reply go.mod:164]: I think you can remove api from this change

File: istio.deps (hunk 2 above)
Comment (jwendell) [reply istio.deps:7]: You can remove proxy from this change. The 1.7 will use another branch for envoy anyway, so this is useless

File: go.mod (hunk 1 above)
Comment (JimmyCYJ) [reply go.mod]: This is reverted. Thanks

File: istio.deps (hunk 2 above)
Comment (JimmyCYJ) [reply istio.deps]: This change is reverted. Thanks

**Review by jwendell — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Comment by jwendell** [comment]
It seems we have no alternative other than wait for https://github.com/istio/istio/pull/25677 to be merged.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**